### ISSUE TITLE
Installation failed when no ttl on OutboxCleanup index

### ISSUE THREAD

**Issue by rvriens** [Particular/NServiceBus.Storage.MongoDB#520]
#### Description
After migration the indexes on all indexes are recreated. The ttl is not a default index you expect.

#### Expected behavior

When no ttl is on the index the index should be recreated.

#### Actual behavior
the installation fails.

#### Versions

Please list the version of the relevant packages or applications in which the bug exists.        
NServiceBus.Storage.MongoDB version 2.3.2

### Steps to reproduce

- Use outbox on mongocollection. Start installation.
- Remove the ttl on the outboxrecord. (remove current OutboxCleanup index, recreate the index without the ttl)
- Start installation of the outbox. This step will fail.

### Relevant log output

```shell
Unhandled exception. System.Collections.Generic.KeyNotFoundException: Element 'expireAfterSeconds' not found.
at MongoDB.Bson.BsonDocument.GetElement(String name)
at NServiceBus.Storage.MongoDB.OutboxStorage.Setup(FeatureConfigurationContext context) in /_/src/NServiceBus.Storage.MongoDB/Outbox/OutboxStorage.cs:line 60
at NServiceBus.Features.FeatureActivator.FeatureInfo.InitializeFrom(FeatureConfigurationContext featureConfigurationContext) in /_/src/NServiceBus.Core/Features/FeatureActivator.cs:line 225
at NServiceBus.Features.FeatureActivator.ActivateFeature(FeatureInfo featureInfo, List`1 featuresToActivate, FeatureConfigurationContext featureConfigurationContext) in /_/src/NServiceBus.Core/Features/FeatureActivator.cs:line 188
at NServiceBus.Features.FeatureActivator.SetupFeatures(FeatureConfigurationContext featureConfigurationContext) in /_/src/NServiceBus.Core/Features/FeatureActivator.cs:line 54
at NServiceBus.FeatureComponent.Initalize(FeatureConfigurationContext featureConfigurationContext) in /_/src/NServiceBus.Core/Features/FeatureComponent.cs:line 29
at NServiceBus.EndpointCreator.Initialize() in /_/src/NServiceBus.Core/EndpointCreator.cs:line 59
at NServiceBus.EndpointCreator.Create(SettingsHolder settings, Configuration hostingConfiguration) in /_/src/NServiceBus.Core/EndpointCreator.cs:line 25
at NServiceBus.HostCreator.CreateWithExternallyManagedContainer(EndpointConfiguration endpointConfiguration, IConfigureComponents externalContainer) in /_/src/NServiceBus.Core/Hosting/HostCreator.cs:line 34
at NServiceBus.EndpointWithExternallyManagedContainer.Create(EndpointConfiguration configuration, IConfigureComponents configureComponents) in /_/src/NServiceBus.Core/EndpointWithExternallyManagedContainer.cs:line 20
at NServiceBus.EndpointWithExternallyManagedServiceProvider.Create(EndpointConfiguration endpointConfiguration, IServiceCollection serviceCollection) in /_/src/NServiceBus.Extensions.DependencyInjection/EndpointWithExternallyManagedServiceProvider.cs:line 21
```

#### Workarounds
Remove the index or remove the collection and restart the installation

#### Possible solutions

Add null check and also recreate index when ttl is null.

**Comment by danielmarbach** [comment]
@rvriens thanks for reporting which such great level of details! 

I have added a reproduction test and a fix in https://github.com/Particular/NServiceBus.Storage.MongoDB/pull/521

**Comment by danielmarbach** [comment]
@rvriens The fix has been merged to our main branch. It's eligible to be picked up for a future release, but we can't provide a firm timeline.

**Comment by danielmarbach** [comment]
@rvriens given there is a workaround and to be exposed to the bug you need to go through a migration, we think we are going to port it only to the 3.0 release version. Do you see any problem with that for your scenario, since you are on the 2.3.x version?

**Comment by rvriens** [comment]
Main is fine. We do use the workaround. Thanks.

**Comment by danielmarbach** [comment]
@rvriens [we have patched 3.0.2](https://discuss.particular.net/t/nservicebus-storage-mongodb-3-0-2-patch-release-available/3634)
